src/gleam/string.gleam

```diff
@@ -36,22 +71,335 @@ pub fn replace(
+///
+/// > right("Scully", 2) == "ly"
+/// 
+// pub fn right(from string: String, num_characters: Int) -> String {}
+
+
+/// Drop *n* characters from the left side of a 
+///
+/// > drop_left(from: "The Lone Gunmen", num_characters: 2) == "e Lone Gunmen"
+/// 
+// pub fn drop_left(from string: String, num_characters: Int) -> String {}
+
+
+/// Drop *n* characters from the right side of a 
+///
+/// > drop_right("Cigarette Smoking Man", 2) == "Cigarette Smoking M"
+/// 
+// pub fn drop_right(from string: String, num_characters: Int) -> String {}
+
+
+
+
+/// ## Check for Substrings
+
+
+
+
+/// See if the second string contains the first one.
+///
+/// > contains("theory", this: "the") == True
+/// > contains("theory", this: "hat") == False
+/// > contains("theory", this: "THE") == False
+///
+// pub fn contains(this: String, in: String) -> String {}
+
+
+/// See if the second string starts with the first one.
+///
+/// > starts_with("theory", this: "the") == True
+/// > starts_with("theory", this: "ory") == False
+///
+// pub fn starts_with(this: String, in: String) -> String {}
+
+
+/// See if the second string ends with the first one.
+///
+/// > endsWith("theory", this: "the") == False
+/// > endsWith("theory", this: "ory") == True
+/// 
+// pub fn ends_with(this: String, in: String) -> String {}
+
+
+
+
+/// ## Building and Splitting
+
+
+
+
+/// Split a string using a given separator.
+///
+/// > split("cat,dog,cow", on: ",")        == ["cat","dog","cow"]
+/// > split("home/evan/Desktop/", on: "/") == ["home","evan","Desktop", ""]
+/// 
+pub fn split(string x: String, on pattern: String) -> List(String) {
+  x
+  |> iodata.new
+  |> iodata.split(_, on: pattern)
+  |> list.map(_, with: iodata.to_string)
+}
+
+
+/// Append two strings.
+///
+/// > append(to: "butter", suffix: "fly") == "butterfly"
+///
 pub fn append(to first: String, suffix second: String) -> String {
   first
   |> iodata.new
   |> iodata.append(_, second)
   |> iodata.to_string
 }
 
+
+/// Concatenate many strings into one.
+///
+/// > concat(["never","the","less"]) == "nevertheless"
+///
 pub fn concat(strings: List(String)) -> String {
   strings
   |> iodata.from_strings
   |> iodata.to_string
 }
 
+
+/// Put many strings together with a given separator.
+///
+/// > join(["H","w","ii","n"], with: "a")        == "Hawaiian"
+/// > join(["cat","dog","cow"], with: " ")       == "cat dog cow"
+/// > join(["home","evan","Desktop"], with: "/") == "home/evan/Desktop"
+///
 pub fn join(strings: List(String), with separator: String) -> String {
   strings
   |> list.intersperse(_, with: separator)
   |> iodata.from_strings
   |> iodata.to_string
 }
+
+
+
+/// Break a string into words, splitting on chunks of whitespace.
+///
+/// > words("How are \t you? \n Good?") == ["How","are","you?","Good?"]
+/// 
+// pub fn words(string: String) -> List(String) {}
+
+
+/// Break a string into lines, splitting on newlines.
+///
+/// > lines("How are you?\nGood?") == ["How are you?", "Good?"]
```

Comment:
`>` is markdown for a blockquote, could we instead use the syntax for a code block? Thanks

Reply:
Done